src/tests/CMakeLists.txt

```diff
@@ -55,10 +55,24 @@ blt_add_executable(
   SOURCES managed_array_tests.cpp
   DEPENDS_ON ${managed_array_test_depends})
 
+blt_add_executable(
+  NAME managed_array_bounds_check_tests
```

Comment:
I'm not clear on the benefit of the extra file? These tests can just be guarded in the main file.

Reply:
But then the tests are only run if the option was specified to cmake...which isn't a big deal, but it also means that to enable the tests in travis, you would need to add an extra ci test. This way the bounds checking is always run for every ci test case.

Reply:
But they should only be run if the CMake option is on, otherwise the bounds are not checked...